mempool/cat/store.go

```diff
@@ -38,13 +38,20 @@ func (s *store) set(wtx *wrappedTx) bool {
 func (s *store) get(txKey types.TxKey) *wrappedTx {
 	s.mtx.RLock()
 	defer s.mtx.RUnlock()
-	return s.txs[txKey]
+	tx := s.txs[txKey]
+	if tx != nil && tx.isReserved() {
+		return nil
+	}
```

Comment:
[Question] In which cases a reserved tx exists, while the original tx does not?

Reply:
if its currently being processed by the mempool

Reply:
Does it mean then there are situations in which the original tx is removed from the mempool, but the reserved one is still in the mempool?

Reply:
The original tx and the reserved tx are the same barring the fact that the height we received the transaction has not yet been set.

If a transaction fails to be added it is removed. If it succeeds the height is added and it's no longer considered reserved